samples/snippets/src/main/java/com/example/firestore/beam/ExampleFirestoreBeamRead.java

```diff
@@ -0,0 +1,128 @@
+                  @ProcessElement
+                  public void processElement(ProcessContext c) {
+                    c.output(Objects.requireNonNull(c.element()).getDocument().getName());
+                  }
+                }))
+        .apply(
+            ParDo.of(
+                new DoFn<String, Void>() {
+                  @ProcessElement
+                  public void processElement(ProcessContext c) {
+                    System.out.println(c.element());
+                  }
+                }));
+
+    pipeline.run().waitUntilFinish();
+  }
+
+  private static final class ReadDocumentsQuery
+      extends PTransform<PCollection<String>, PCollection<RunQueryRequest>> {
+
+    private final String projectId;
+    private final String databaseId;
+
+    public ReadDocumentsQuery(String projectId, String databaseId) {
+      this.projectId = projectId;
+      this.databaseId = databaseId;
+    }
+
+    @Override
+    public PCollection<RunQueryRequest> expand(PCollection<String> input) {
+      return input.apply(
+          ParDo.of(
+              new DoFn<String, RunQueryRequest>() {
+
+                @ProcessElement
+                public void processElement(ProcessContext c) {
+                  RunQueryRequest runQueryRequest =
+                      RunQueryRequest.newBuilder()
+                          .setParent(DocumentRootName.format(projectId, databaseId))
+                          .setStructuredQuery(
+                              StructuredQuery.newBuilder()
+                                  .addFrom(
+                                      StructuredQuery.CollectionSelector.newBuilder()
+                                          .setCollectionId(Objects.requireNonNull(c.element()))
+                                          .build())
+                                  .setWhere(
+                                      StructuredQuery.Filter.newBuilder()
+                                          .setFieldFilter(
+                                              StructuredQuery.FieldFilter.newBuilder()
+                                                  .setField(
+                                                      StructuredQuery.FieldReference.newBuilder()
+                                                          .setFieldPath("country")
+                                                          .build())
+                                                  .setValue(
+                                                      Value.newBuilder()
+                                                          .setStringValue("USA")
+                                                          .build())
+                                                  .setOp(
+                                                      StructuredQuery.FieldFilter.Operator.EQUAL))
+                                          .buildPartial())
+                                  .build())
+                          .build();
```

Comment:
This nesting seems to be very deep. Is this a general okay pattern according to our java style guide? I was trying to look up something around it but couldn't find anything. If it is not specifically an accepted guide can we split this into separate blocks?

 ```java
Function<ProcessContext, StructuredQuery.CollectionSelector> selectorFn = (c) -> {
  return StructuredQuery.CollectionSelector.newBuilder()
      .setCollectionId(Objects.requireNonNull(c.element()))
      .build();
};
StructuredQuery.FieldReference countryFieldRef = StructuredQuery.FieldReference.newBuilder()
    .setFieldPath("country")
    .build();
Value countryValueRef = Value.newBuilder()
    .setStringValue("USA")
    .build();
StructuredQuery.FieldFilter.Builder countryFieldFilterBuilder = StructuredQuery.FieldFilter.newBuilder()
    .setField(countryFieldRef)
    .setValue(countryValueRef)
    .setOp(StructuredQuery.FieldFilter.Operator.EQUAL);
StructuredQuery.Filter countryFilter = StructuredQuery.Filter.newBuilder()
    .setFieldFilter(countryFieldFilterBuilder)
    .buildPartial();

@ProcessElement
public void processElement(ProcessContext c) {
  RunQueryRequest runQueryRequest = RunQueryRequest.newBuilder()
      .setParent(DocumentRootName.format(projectId, databaseId))
      .setStructuredQuery(
          StructuredQuery.newBuilder()
              .addFrom(selectorFn.apply(c))
              .setWhere(countryFilter)
              .build())
      .build();
  c.output(runQueryRequest);
}
```

Reply:
great suggestion, thanks!